apps/ledger-live-mobile/src/families/evm/EvmFeesStrategy.tsx

```diff
@@ -104,7 +132,7 @@ export default function EvmFeesStrategy({
     <SelectFeesStrategy
       {...props}
       gasOptions={gasOptions}
-      customFees={customStrategy}
+      customFees={customStrategyFees}
```

Comment:
We could simplify a bit the logic and derive the `customFees` value from the saved patch without having to store it in a new state variable by going with this suggestion.
The only minor issue is that we have to cast the object as a `Transaction` because it looks like TS complains about a compatibility issue between `transaction` and `customStrategyTransactionPatch` modes. Which is weird since `transaction` is of type `Transaction` and `customStrategyTransactionPatch` is of type `Partial<Transaction>` 🤔 

```
          Types of property 'mode' are incompatible.
            Type '"send" | "erc721" | "erc1155"' is not assignable to type '"erc721" | "erc1155"'.
              Type '"send"' is not assignable to type '"erc721" | "erc1155"'.
```

```suggestion
      customFees={getCustomStrategyFees({
        ...transaction,
        ...customStrategyTransactionPatch,
      } as Transaction)}
```

Reply:
Calling a method like this is viewed as bad practice in Angular because it causes rerenders for no reason, is it fine in React ?

To answer to your question maybe it is just a type error, we can see on one side that the type isn't complete and misses 'send' value

since one overrides the other i think it should be fine but you should check why it's happening

Reply:
if you use `Pick<Transaction, "feesStrategy">` insead of `Partial<Transaction>` this will work. I agree the error shouldn't be raise but it kinda make sense

Reply:
> Calling a method like this is viewed as bad practice in Angular because it causes rerenders for no reason, is it fine in React ?

I don't see how it would cause rerenders.
It would cause unnecessary rerenders if it was an action function updating state value. Which is not the case here.

This function is just formatting value (more precisely, returning a computation of 2 values from `transaction`, or `null`).

One could decide to put this computation in a `useMemo` like so:

```ts
  const customFees = useMemo(
    () =>
      getCustomStrategyFees({
        ...transaction,
        ...customStrategyTransactionPatch,
      } as Transaction),
    [transaction, customStrategyTransactionPatch],
  );

....

  return (
    <SelectFeesStrategy
      customFees={customFees}
      ....
    />
```

But this is not necessary in this case since the memoised computation is not expensive. The cost of memoisation would probably exceed any "gain" (if any, since the memoised value would be invalidated and trigger a rerender each time `transaction` or `customStrategyTransactionPatch` anyway)
cf. https://react.dev/reference/react/useMemo#should-you-add-usememo-everywhere

> if you use Pick<Transaction, "feesStrategy"> insead of Partial<Transaction> this will work. I agree the error shouldn't be raise but it kinda make sense

I guess you are talking about this type:

```ts
  const [customStrategyTransactionPatch, setCustomStrategyTransactionPatch] =
    useState<Partial<Transaction>>();
```

But `customStrategyTransactionPatch` is not of type `feesStrategy`, it also includes other props of a transaction  related to fees (cf. where `setCustomStrategyTransactionPatch` is used under `apps/ledger-live-mobile/src/families/evm/EvmCustomFees`